include/openenclave/attestation/custom_claims.h

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifndef _OE_CUSTOM_CLAIMS
+#define _OE_CUSTOM_CLAIMS
+
+#include <openenclave/bits/evidence.h>
+#include <openenclave/bits/result.h>
+
+OE_EXTERNC_BEGIN
+#define OE_CUSTOM_CLAIMS_VERSION 1
+
+typedef struct _oe_custom_claims_header
+{
+    uint64_t version;
+    uint64_t num_claims;
+} oe_custom_claims_header_t;
+
+/**
+ * Serialized entry for custom claims. Each entry will have the name and value
+ * sizes and then the contents of the name and value respectively.
+ */
+typedef struct _oe_custom_claims_entry
+{
+    uint64_t name_size;
+    uint64_t value_size;
+    uint8_t name[];
+    // name_size bytes follow.
+    // value_size_bytes follow.
+} oe_custom_claims_entry_t;
+
+/**
+ * oe_serialize_custom_claims
+ *
+ * Serializes a list of custom claims.
+ *
+ * @param[in] custom_claims Custom claims to serialize.
+ * @param[in] custom_claims_length Length of custom_claims.
+ * @param[out] claims_out Pointer to the address of a dynamically
+ * allocated buffer holding the serialized custom claims.
+ * @param[out] claims_size_out Size of the serialized custom claims.
+ * @retval OE_OK on success.
+ * @retval OE_INVALID_PARAMETER At least one parameter is invalid.
+ * @retval An appropriate error code on failure.
+ */
+oe_result_t oe_serialize_custom_claims(
+    const oe_claim_t* custom_claims,
+    size_t custom_claims_length,
+    uint8_t** claims_out,
+    size_t* claims_size_out);
+
+/**
+ * oe_deserialize_custom_claims
+ *
+ * Deserialize custom claim buffer
+ *
+ * @param[in] claims_buffer Pointer to the serialized custom claims buffer.
+ * @param[in] claims_buffer_size Size of the serialized custom claims buffer
+ * buffer.
+ * @param[out] claims_out Pointer to the address of a dynamically allocated
+ * buffer holding the list of custom claims.
+ * @param[out] claims_length_out The length of the claims_out list.
+ * @retval OE_OK on success.
+ * @retval OE_INVALID_PARAMETER At least one parameter is invalid.
+ * @retval An appropriate error code on failure.
+ */
+oe_result_t oe_deserialize_custom_claims(
```

Comment:
Very much agree.  Since you're hiding which allocator is used, we also need to hide which deallocator is used so the caller need not know whether free or something else is the right API.

Reply:
Added oe_free_serialized_custom_claims(), oe_free_custom_claims()